velox/expression/tests/CastExprTest.cpp

```diff
@@ -1173,3 +1173,21 @@ TEST_F(CastExprTest, dictionaryEncodedNestedInput) {
   auto expectedArray = makeArrayVector({0, 3}, expectedRow);
   assertEqualVectors(expectedArray, result);
 }
+
+TEST_F(CastExprTest, smallerNonNullRowsSizeThanRows) {
+  // Evaluating Cast in Coalesce as the second argument triggers the copy of
+  // Cast localResult to the result vector. The localResult vector is of the
+  // size nonNullRows which can be smaller than rows. This test ensures that
+  // Cast doesn't attempt to access values out-of-bound and hit errors.
+  exec::registerVectorFunction(
+      "add_dict",
```

Comment:
nit: would it make sense to rename the function to "add_dict_with_2_trailing_nulls", its a bit verbose, but the test case confused me at first till I looked up TestingDictionaryFunction.